Replace Travis and AppVeyor with GitHub Actions

Midnighter opened the pull request:
* Clean up old files
* Provide new templates
* Update versioneer
* Update setup
* Expand use of tox
* Move all code to `src/optlang`

What's missing:

* Create secrets for deployment to PyPI (need some assistance from @phantomas1234 here)
* Download and install CPLEX as it was on Travis (that's why I didn't remove the `.travis.yml` yet)
* Clean up coverage configuration

Let me know if you think this is going in the right direction.

cdiener commented:
Regarding cplex. There is an official [free test version on PYPI](https://pypi.org/project/cplex/) that can be used as long as you have less than 1000 variables/constraints. If we separate tests into small and large ones this one could be used (hard with the current test setup though).

Midnighter commented:
That's a great idea! (I wish there was something like that for Gurobi, too.) I have some local work that was aimed at optlang 2.0 which introduces a pytest-based test suite. It'd be a good solution to introduce markers for slow and fast, as well as small and large tests.

Midnighter commented:
* I have now fully removed the Travis CI config and dropped support for CPLEX testing for now. I've added an issue for @cdiener 's idea in #218 so I want to add it back in ASAP.
* I've also set up the tokens so we can make PyPI releases again.
* I've temporarily switch the codecov GitHub Action to the Python command line tool.

In my opinion this is ready to be merged.

@phantomas1234 it looks like the swiglpk wheels for macos Python 3.5 are missing.

KristianJensen commented on .github/workflows/main.yml:
```diff
@@ -0,0 +1,87 @@
+on:
+  push:
+    branches:
+    - stable
```
I've never used github workflows before, so I might be missing something. But should this not be `master`?

Midnighter replied:
Absolutely!

KristianJensen commented:
This looks very good to me! I've added a comment on something that might be a mistake, or just me not fully understanding gh_actions.

Also, have you set it up to test with and without symengine like on travis? And if so, (for my understanding) where is that specified?

Midnighter commented:
> Also, have you set it up to test with and without symengine like on travis? And if so, (for my understanding) where is that specified?

Indeed, this is set in the `tox.ini` for Python 2.7 and 3.6.

```ini
[gh-actions]
python =
     2.7: safety, py27, py27-symengine
     3.5: safety, py35
     3.6: safety, py36, py36-symengine
     3.7: safety, py37
     3.8: safety, py38
```

We could run it on more versions (or all of them). It actually doesn't take that long.

Midnighter commented:
Wanted to add that `[gh-actions]` section above is only used to select environments for specific Python versions. This then gets combined with the test environment definition.

```ini
[testenv]
deps =
     colorama ~= 0.3.9
     coverage
     docutils
     jsonschema
     nose
     nose-progressive
     rednose
     scipy
     symengine: symengine
```
where the last line is a selector that installs symengine if symengine is part of the tox environment name.

Feel free to merge this or if you accept it in review I will merge it.

Midnighter commented:
@KristianJensen can we merge this and make a new release or is there something that needs to be changed?

KristianJensen commented:
Looks good to me!
